### PR TITLE
UIROLES-43 Fixes to improve add/remove roles behavior

### PR DESCRIPTION
**Pull request by ryandberger**
- Add error handling to APIs
- Invalidate react-query to improve refetch of assigned users
- Fix POST case

### CODE REVIEW

File: src/settings/components/RoleDetails/utils.js [hunk 1]
```diff
@@ -38,18 +38,18 @@ export function createUserRolesRequests(previousSelectedUsers, currentSelectedUs
...
       // If modified, then PUT
       if ((added.includes(userId) || removed.includes(userId)) && roleIds.length) {
         requests.push({ userId, roleIds, apiVerb: apiVerbs.PUT });
       } else if (!roleIds.length) { // If no more capabilities, DELETE
         requests.push({ userId, roleIds, apiVerb: apiVerbs.DELETE });
       }
-    } else { // if no matches, POST
+    } else if (added?.includes(userId)) { // if no matches, POST
       roleIds.push(roleId);
       requests.push({ userId, roleIds, apiVerb: apiVerbs.POST });
```
Comment (zburke) [inline src/settings/components/RoleDetails/utils.js:54]: More documentation in this section would be great. It was hard for me to follow how the manipulation of `roleIds` related to the various conditions.

File: src/hooks/useAssignRolesToUserMutation.js [hunk 2]
```diff
@@ -9,7 +9,8 @@ const useAssignRolesToUserMutation = () => {
     mutationFn: (newRole) => ky.post('roles/users', { json: newRole }).json(),
     onSuccess: async () => {
       await queryClient.invalidateQueries(namespace);
-    }
+    },
+    onError: (error) => console.error(JSON.stringify(error)) // eslint-disable-line no-console
```
Comment (zburke) [inline src/hooks/useAssignRolesToUserMutation.js:13]: Should we be doing more than just reporting this in the console? e.g. should we actually shove this into an `alert()` for now until legit, user-friendly error handling is implemented?

**Review by zburke — APPROVED** [review]
The changes look good to me and code functions as expected. Yay! But I think we would benefit from improved comments/documentation and better error handling. By trapping errors and writing them to the console, I worry we will mislead/under-inform end-users who will click a button and then wonder, "Did anything happen?" when there's an error that was merely reported on developer console.

File: src/hooks/useAssignRolesToUserMutation.js (hunk 2 above)
Comment (craigmcnally) [reply src/hooks/useAssignRolesToUserMutation.js:13]: I will create a follow-on story to address this.

File: src/settings/components/RoleDetails/utils.js (hunk 1 above)
Comment (craigmcnally) [reply src/settings/components/RoleDetails/utils.js:54]: I'll include this in the follow-on story I'll create for better error handling.

File: src/hooks/useAssignRolesToUserMutation.js (hunk 2 above)
Comment (craigmcnally) [reply src/hooks/useAssignRolesToUserMutation.js:13]: Created https://folio-org.atlassian.net/browse/UIROLES-53 for this.

**Pull request merged**